dev/docker/build-docker.sh

```diff
@@ -108,14 +108,14 @@ fi
 cd ${script_dir}/${component_type}
 if [[ "${platform_type}" == "all" ]]; then
   if [ ${build_latest} -eq 1 ]; then
-    docker buildx build --no-cache --pull --platform=linux/amd64,linux/arm64 ${build_args} --push --progress plain -f Dockerfile -t ${image_name}:latest -t ${image_name}:${tag_name} .
```

Comment:
Why do we need to delete `--no-cache`?

Reply:
Because It may make it hard to tweak `Dockerfile`, with `-no-cache`, while rebuilding Docker image, we needs to take 3-5 min to do `apt-get` commands.

What's the reason of adding this `--no-cache` ?

Reply:
It's likely that some changes to the docker files won't take effect if we reuse the cache, I 've encountered this issue several times. 

> Because It may make it hard to tweak Dockerfile, with -no-cache, while rebuilding Docker image, we needs to take 3-5 min to do apt-get commands.

I believe you can remove it during a local test and then revert it once you commit the code.

Reply:
> It's likely that some changes to the docker files won't take effect if we reuse the cache, I 've encountered this issue several times.

Okay, If this happens, we should not remove `--no-cache`.

Just curious, Could you give me an example of this bug ?

Reply:
You can view the commit ID and obtain information about it.  It's exactly in this PR that I introduced option `--no-cache` 
<img width="1625" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/ab187760-7ac0-473b-8730-51cbfea62afe">

To be honest, I have n't come across this issue in the recent changes to the docker files.

Reply:
Okay, I'll take a look, and we can keep `--no-cache` for now, until we find the root cause.